src/infrastructure/acknowlegement.rs

```diff
@@ -0,0 +1,42 @@
+use crate::net::{ExternalAcks, LocalAckRecord};
+
+/// Type responsible for handling the acknowledgement of packets.
+pub struct AcknowledgementHandler {
+    waiting_packets: LocalAckRecord,
+    their_acks: ExternalAcks,
+    pub seq_num: u16,
+    pub dropped_packets: Vec<Box<[u8]>>,
+}
+
+impl AcknowledgementHandler {
+    pub fn new() -> AcknowledgementHandler {
+        AcknowledgementHandler {
+            seq_num: 0,
+            waiting_packets: Default::default(),
+            their_acks: Default::default(),
+            dropped_packets: Vec::new(),
+        }
+    }
+}
+
+impl AcknowledgementHandler {
```

Comment:
Can we get /// comments for all the pub functions, please?

Reply:
I see why you want that, I left it away because it is not public for the user.

Reply:
done